tools/swift-format/Sources/Rules/NoAccessLevelOnExtensionDeclaration.swift

```diff
@@ -10,7 +10,148 @@ import SwiftSyntax
 public final class NoAccessLevelOnExtensionDeclaration: SyntaxFormatRule {
+  
+  public override func visit(_ node: ExtensionDeclSyntax) -> DeclSyntax {
+    guard let modifiers = node.modifiers, modifiers.count != 0 else { return node }
+    
+    var modifierIterator = modifiers.makeIterator()
+    if let accessKeyword = modifierIterator.next() {
+
+      let keywordKind = accessKeyword.name.tokenKind
+      switch keywordKind {
+      // Public, private, or fileprivate keywords need to be moved to members
+      case .publicKeyword, .privateKeyword, .fileprivateKeyword:
+        diagnose(.moveAccessKeyword(keyword: accessKeyword.name.text), on: accessKeyword)
+        let newMembers = SyntaxFactory.makeMemberDeclBlock(
+          leftBrace: node.members.leftBrace,
+          members: addMemberAccessKeywords(memDeclBlock: node.members, keyword: accessKeyword),
+          rightBrace: node.members.rightBrace)
+        return node.withMembers(newMembers)
+                .withModifiers(removeModifier(curModifiers: modifiers, removal: accessKeyword))
+      // Internal keyword redundant, delete
+      case .internalKeyword:
+        diagnose(.removeRedundantAccessKeyword(name: node.extendedType.description), on: accessKeyword)
+        return node.withModifiers(removeModifier(curModifiers: modifiers, removal: accessKeyword))
+                .withExtensionKeyword(node.extensionKeyword.withOneLeadingNewline())
+      default:
+        return node
+      }
+    }
+    return node
+  }
+  
+  // Returns modifier list without the access modifier
+  func removeModifier(curModifiers: ModifierListSyntax, removal: DeclModifierSyntax) -> ModifierListSyntax {
+    var newMods: [DeclModifierSyntax] = []
+    for modifier in curModifiers {
+      if modifier.name != removal.name {
+        newMods.append(modifier)
+      }
+    }
+    return SyntaxFactory.makeModifierList(newMods)
+  }
+  
+  // Adds given keyword to all members in declaration block
+  func addMemberAccessKeywords(memDeclBlock: MemberDeclBlockSyntax, keyword: DeclModifierSyntax) -> MemberDeclListSyntax {
+    var newMembers: [MemberDeclListItemSyntax] = []
+    let formattedKeyword = keyword.withName(keyword.name.withLeadingTrivia(.newlines(1) + .spaces(2)))
+    
+    for member in memDeclBlock.members {
+      
+      if let varDecl = member.decl as? VariableDeclSyntax {
+        // Check for modifiers, if none, put accessor keyword before var/let keyword
+        guard let modifiers = varDecl.modifiers
+          else {
+            let formattedVarDecl = varDecl.withLetOrVarKeyword(varDecl.letOrVarKeyword.withoutLeadingTrivia())
+            newMembers.append(member.withDecl(formattedVarDecl.addModifier(formattedKeyword)))
+            continue
+          }
+        // If variable already has an accessor keyword, skip (do not overwrite)
+        guard hasAccessorKeyword(modifiers: modifiers) == false
+          else { newMembers.append(member); continue }
```

Comment:
Formatting nit: I think we prefer the `else` on the same line, right @allevato?

Reply:
Also, we prefer one statement per line, so if you can split `newMembers.append(member)` and `continue` that'd be great!

Reply:
If it fits with the rest of the line, yes; so this would be:

```swift
guard hasAccessorKeyword(modifiers: modifiers) == false else {
  newMembers.append(member)
  continue
}
```

(I'm ambivalent about whether it should be `hasAccessorKeyword(...) == false` or `!hasAccessorKeyword(...)`. Our style guide doesn't care.)